### PR TITLE
UPBGE: Implement SMAA following workbench code

### PR DESCRIPTION
**Pull request by youle31**
This works both in viewport and bge

### CODE REVIEW

File: source/blender/draw/engines/eevee/eevee_effect_antialiasing.c [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+  else {
+    /* Accumulate result to the TAA buffer. */
+    GPU_framebuffer_bind(fbl->antialiasing_fb);
+    DRW_draw_pass(psl->aa_accum_ps);
+    /* Copy back the saved depth buffer for correct overlays. */
+    GPU_framebuffer_blit(fbl->antialiasing_fb, 0, dfbl->default_fb, 0, GPU_DEPTH_BIT);
+  }
+
+  /* After a certain point SMAA is no longer necessary. */
+  if (vedata->stl->effects->taa_current_sample < 16) {
```
Comment (lordloki) [inline source/blender/draw/engines/eevee/eevee_effect_antialiasing.c]: This ´´´if (vedata->stl->effects->taa_current_sample < 16)´´´ check is similar to check from line 337. Could they unified in the check above? (and i think that the comment should put just above smaa_mix_factor definition not above ´´´if (vedata->stl->effects->taa_current_sample < 16)´´´ check)

Comment (lordloki) [reply source/blender/draw/engines/eevee/eevee_effect_antialiasing.c]: Other than this it seems Ok. I'm going to test it. It is exciting!!!! ,😁👍

Comment (youle31) [reply source/blender/draw/engines/eevee/eevee_effect_antialiasing.c]: hello! thanks for review, i just pushed 2 or 3 more cleanup commits. Thanks very much for review and happy testing. I hope this won't crash : )

**Comment by lordloki** [comment]
It works OK in my test files. Good work!!!!

**Comment by youle31** [comment]
Thanks very much! I merge

**Pull request merged**